### PR TITLE
fix for preflight requests

### PR DESCRIPTION
**Pull request by vastbinderj**
handle preflight requests

### CODE REVIEW

**Comment by yosisa** [comment]
+1 I have the same issue.

File: cors.go [hunk 1]
```diff
@@ -192,10 +192,13 @@ func Allow(opts *Options) http.HandlerFunc {
 			(requestedMethod != "" || requestedHeaders != "") {
 			// TODO: if preflight, respond with exact headers if allowed
 			headers = opts.PreflightHeader(origin, requestedMethod, requestedHeaders)
+			for key, value := range headers {
+				res.Header().Set(key, value)
+			}
 			res.WriteHeader(http.StatusOK)
-		} else {
-			headers = opts.Header(origin)
+			return
```
Comment (rakyll) [inline cors.go:199]: I didn't understand why we return here? Shouldn't the flow be like:

``` code
if (preflight) 
  set preflight headers
  set HTTP_OK
  return

set regular cors headers
pass it to the next handler (martini does that if status is not set)
```

Comment (vastbinderj) [inline cors.go:199]: That should work, but it wasn't.  I had to insert the return to interrupt Martini and send the OPTIONS headers.

Comment (rakyll) [inline cors.go:199]: I was baffled around an another issue due to the diff, please ignore my previous comment. Merging now.

**Pull request merged**